spec/immutable-data-structures.html

```diff
@@ -49,8 +49,10 @@ <h1>Record.fromEntries ( _iterable_ )</h1>
           1. Perform ? RequireObjectCoercible(_iterable_).
           1. Let _fields_ be a new empty List.
           1. Let _adder_ be a new Abstract Closure with parameters (_key_, _value_) that captures _fields_ and performs the following steps when called:
+            1. If Type(_key_) is Symbol, throw a *TypeError* exception.
             1. If Type(_value_) is Object, throw a *TypeError* exception.
-            1. Let _field_ be { [[Key]]: _key_, [[Value]]: _value_ }.
+            1. Let _keyString_ be ? ToPropertyKey(_key_)
```

Comment:
How can ToPropertyKey throw if you know key isn’t an object?

Reply:
Then this call could produce a Symbol, so the symbol check must be done after this line.

Reply:
great spot!

Reply:
I've gone back to `ToString` this time without the extra superfluous check for symbol.

I am now also wondering if we should do the key processing before throwing if the value is not an object. As the key is _earlier_ in the array.

Reply:
Yes, 1000%

Reply:
done!